In [1]:
import mysql.connector
import pandas as pd

# fisrt data source is from mysql

In [2]:
connection = mysql.connector.connect(host = 'localhost',user = 'root', password = 'cUny$p$_7', db ='college' )

In [3]:
query = '''SELECT institution, world_rank, national_rank, country, year FROM world_university_rankings WHERE country="USA"'''

In [4]:
rank = pd.read_sql_query(query, connection)

In [5]:
rank.rename(columns={'institution':'college_id'}, inplace=True)

# convert column in to date time format.

In [6]:
rank['year'] = pd.to_datetime(rank['year'], format='%Y')

In [7]:
rank.dtypes

college_id               object
world_rank                int64
national_rank             int64
country                  object
year             datetime64[ns]
dtype: object

In [8]:
rank.head()

college_id  world_rank  national_rank country  \
0                     Harvard University           1              1     USA   
1                    Stanford University           2              2     USA   
2  Massachusetts Institute of Technology           3              3     USA   
3                    Columbia University           6              4     USA   
4     University of California, Berkeley           7              5     USA   

        year  
0 2014-01-01  
1 2014-01-01  
2 2014-01-01  
3 2014-01-01  
4 2014-01-01

# execute second query to get the number of ranked univeristies in the USA.

In [9]:
query = '''SELECT country, count(*) FROM world_university_rankings WHERE country="USA" GROUP BY country'''

In [10]:
rank_colleges_in_USA = pd.read_sql_query(query, connection)

In [11]:
rank_colleges_in_USA

country  count(*)
0     USA       458

# There are 458 USA Universities in rank on the world!

# The second data source from GitHub

In [12]:
Tuition = pd.read_csv("https://raw.githubusercontent.com/harvey11230/Final-Project/master/National%20Universities%20Rankings.csv")

In [13]:
Tuition.head()

Name       City State  \
0   Princeton University  Princeton    NJ   
1     Harvard University  Cambridge    MA   
2  University of Chicago    Chicago    IL   
3        Yale University  New Haven    CT   
4    Columbia University   New York    NY   

                                         Description Tuition and fees  \
0  Princeton, the fourth-oldest college in the Un...         $45,320    
1  Harvard is located in Cambridge, Massachusetts...         $47,074    
2  The University of Chicago, situated in Chicago...         $52,491    
3  Yale University, located in New Haven, Connect...         $49,480    
4  Columbia University, located in Manhattan's Mo...         $55,056    

  In-state Undergrad Enrollment  
0      NaN                5,402  
1      NaN                6,699  
2      NaN                5,844  
3      NaN                5,532  
4      NaN                6,102

# replace the NaN records to 0

In [14]:
Tuition = Tuition.fillna(0).head()

In [15]:
Tuition.rename(columns={'Name':'college_id'}, inplace=True)

In [16]:
Tuition.head()

college_id       City State  \
0   Princeton University  Princeton    NJ   
1     Harvard University  Cambridge    MA   
2  University of Chicago    Chicago    IL   
3        Yale University  New Haven    CT   
4    Columbia University   New York    NY   

                                         Description Tuition and fees  \
0  Princeton, the fourth-oldest college in the Un...         $45,320    
1  Harvard is located in Cambridge, Massachusetts...         $47,074    
2  The University of Chicago, situated in Chicago...         $52,491    
3  Yale University, located in New Haven, Connect...         $49,480    
4  Columbia University, located in Manhattan's Mo...         $55,056    

  In-state Undergrad Enrollment  
0        0                5,402  
1        0                6,699  
2        0                5,844  
3        0                5,532  
4        0                6,102

# Join 2 tables in to 1 to display the world rank, national rank and tuition.

In [17]:
Result = pd.merge(rank,Tuition, on='college_id' , how='inner')

In [18]:
Result = Result.drop_duplicates(subset='college_id',keep='first')

In [19]:
Result.shape

(5, 11)

In [20]:
Result[['college_id', 'world_rank', 'national_rank','City',
       'State', 'Tuition and fees', 'In-state',
       'Undergrad Enrollment','year']]

college_id  world_rank  national_rank       City State  \
0     Harvard University           1              1  Cambridge    MA   
2    Columbia University           6              4   New York    NY   
4  University of Chicago           8              6    Chicago    IL   
6   Princeton University           9              7  Princeton    NJ   
8        Yale University          10              8  New Haven    CT   

  Tuition and fees In-state Undergrad Enrollment       year  
0         $47,074         0                6,699 2014-01-01  
2         $55,056         0                6,102 2014-01-01  
4         $52,491         0                5,844 2014-01-01  
6         $45,320         0                5,402 2014-01-01  
8         $49,480         0                5,532 2014-01-01

# Conlcusion: As result, I found 5 ranked universities with corrsonding tuition and fee from the database.